python/uv/private/uv.bzl

```diff
@@ -18,36 +18,366 @@ EXPERIMENTAL: This is experimental and may be removed without notice
+                platform = default_attr.platform,
+                compatible_with = default_attr.compatible_with,
+                target_settings = default_attr.target_settings,
+            )
+
+    versions = {}
+    for mod in module_ctx.modules:
+        last_version = None
+        for config_attr in mod.tags.configure:
+            last_version = config_attr.version or last_version or defaults["version"]
+            if not last_version:
+                fail("version must be specified")
+
+            specific_config = versions.setdefault(
+                last_version,
+                {
+                    "base_url": defaults.get("base_url", ""),
+                    "manifest_filename": defaults["manifest_filename"],
+                    "platforms": dict(defaults["platforms"]),  # copy
+                },
+            )
+
+            _configure(
+                specific_config,
+                base_url = config_attr.base_url,
+                manifest_filename = config_attr.manifest_filename,
+                platform = config_attr.platform,
+                compatible_with = config_attr.compatible_with,
+                target_settings = config_attr.target_settings,
+                sha256 = config_attr.sha256,
+                urls = config_attr.urls,
             )
 
+    versions = {
+        version: config
+        for version, config in versions.items()
+        if config["platforms"]
+    }
+    if not versions:
+        return struct(
+            names = ["none"],
+            labels = {
+                # NOTE @aignas 2025-02-24: the label to the toolchain can be anything
+                "none": str(Label("//python:none")),
+            },
+            compatible_with = {
+                "none": ["@platforms//:incompatible"],
+            },
+            target_settings = {},
+        )
+
+    toolchain_names = []
+    toolchain_labels_by_toolchain = {}
+    toolchain_compatible_with_by_toolchain = {}
+    toolchain_target_settings = {}
+    for version, config in versions.items():
+        platforms = config["platforms"]
+
+        # Use the manually specified urls
+        urls = {
+            platform: src
+            for platform, src in config.get("urls", {}).items()
+            if src.urls
+        }
+
+        # Or fallback to fetching them from GH manifest file
+        # Example file: https://github.com/astral-sh/uv/releases/download/0.6.3/dist-manifest.json
+        if not urls:
+            urls = _get_tool_urls_from_dist_manifest(
+                module_ctx,
+                base_url = "{base_url}/{version}".format(
+                    version = version,
+                    base_url = config["base_url"],
+                ),
+                manifest_filename = config["manifest_filename"],
+                platforms = sorted(platforms),
+            )
+
+        result = uv_repositories(
+            name = "uv",
+            platforms = {
```

Comment:
This is passing a dict, but it looks like uv_repositories is expecting a list? I don't see the values of the dict used.